### DISCUSSION TITLE
How to add an array of parameters to --kubelet-args

### DISCUSSION THREAD

**Discussion by mscott83** [k3s-io/k3s#5524] [Q&A]
I want to add some kubelet args as part of my K3S node installation process so that Longhorn storage works correctly on my Flatcar hosts

These are the parameters I want to add: 

```
kubelet:
     extra_binds:
      - /usr/sbin/iscsiadm:/usr/sbin/iscsiadm
      - /usr/sbin/iscsid:/usr/sbin/iscsid
      - /etc/iscsi/:/etc/iscsi/
```
    
How do I do this within the environment variables for the installation process? I've tried a few variations of this but none work:

`INSTALL_K3S_EXEC=server --cluster-init  --kubelet-arg='extra_binds=/usr/sbin/iscsiadm:/usr/sbin/iscsiadm,/usr/sbin/iscsid:/usr/sbin/iscsid,/etc/iscsi/:/etc/iscsi/'`

The documentation is pretty weak in this area, being devoid of examples. Any pointers?

**Comment by brandond** [comment]
The k3s kubelet does not run in container, so the concept of adding additional bind-mounts to it does not make any sense. Are you perhaps trying to adapt instructions from a different Kubernetes distro?

**Reply by mscott83** [reply]
...quite possibly.

To broaden the question - does anyone know how to get Longhorn storage working within K3S on Flatcar? Everything installs fine but whenever I create a PVC/PV, it just keeps attaching then detaching and ultimately the storage is unusable. I'm migrating from K3OS where this was never an issue.

**Reply by brandond** [reply]
Do the logs contain any clues as to what's going on? Are you perhaps missing the iscsi bits? You might be better off gathering some logs and taking your issue to the longhorn repo.
